crates/duty-tracker/src/contract_manager.rs

```diff
@@ -553,15 +570,24 @@ impl ContractManagerCtx {
         let deposit_idx_offset = self.state.stake_chains.height();
 
         for tx in block.txdata {
+            let txid = tx.compute_txid();
+
+            let tx_process_start_time = Instant::now();
             // could be an assignment
             let assignment_duties = self.process_assignments(&tx).await?;
             if !assignment_duties.is_empty() {
                 info!(num_duties=%assignment_duties.len(), "queueing assignment duties");
                 duties.extend(assignment_duties);
+                trace!(time_taken=?tx_process_start_time.elapsed(), "processed assignments for {txid}");
+
+                // It's impossible for a transaction to contain assignments and otherwise have any
+                // effect on existing or new contracts, so we move on.
+                continue;
             }
 
             let txid = tx.compute_txid();
```

Comment:
```suggestion
```
the merge main commit made this appear twice -- already defined above just before the if clause.